### ISSUE TITLE
Add support for dynamic metadata in http ext authz

### ISSUE THREAD

**Issue by nothinux** [envoyproxy/gateway#4163]
*Description*:
We have a use case that requires emitting specific information from the external authz service into the logs, and this can be achieved using dynamic metadata. In the grpc ext_authz, dynamic metadata can be included directly as part of `CheckResponse`. However, in the http ext_authz, the only method to pass this information is by sending it as response headers, which are then emitted via the `dynamic_metadata_from_headers`.

It would be great if the [dynamic_metadata_from_headers](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto#extensions-filters-http-ext-authz-v3-authorizationresponse) field can be exposed

**Comment by arkodg** [comment]
can this be achieved today by `headersToBackend` and then removing the header using `RequstHeaderModifier` filter https://gateway-api.sigs.k8s.io/guides/http-header-modifier/#http-request-header-modifier?

**Comment by nothinux** [comment]
@arkodg  yes, it can be archieved with that approach. However, we won't be able to retrieve the [ext_authz_duration](ext_authz_duration) data, since it's only available when `dynamic_metadata_from_headers` is set. We need that data to monitor the time taken to complete the authorization process for each request

**Comment by arkodg** [comment]
this feels like an advanced use case, I suggest using https://gateway.envoyproxy.io/docs/tasks/extensibility/envoy-patch-policy/ to achieve this
lets keep this issue open to see if others in the community are also interested in this

**Comment by arkodg** [comment]
we are beginning to add first class support for dynamic metadata
https://github.com/envoyproxy/gateway/pull/5023 added the ability to read/write metadata for ext proc, we can follow something similar here for ext authz

**Comment by arkodg** [comment]
cc @nareddyt
